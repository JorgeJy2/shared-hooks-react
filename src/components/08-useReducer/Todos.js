import React from 'react'
import PropTypes from 'prop-types';
import { Todo } from './Todo';

export const Todos = ({ todos, handlerToggle, handlerDelete }) => {
    return (
        <ul className="list-group list-group-flush">
            {
                todos.map((todo, i) => <Todo 
                    key={todo.id}
                    todo={todo}
                    index={i}
                    handlerDelete={handlerDelete}
                    handlerToggle={handlerToggle} />)
            }
        </ul>
    )
}
Todos.propTypes = {
    todos: PropTypes.array.isRequired,
    handlerToggle: PropTypes.func.isRequired,
    handlerDelete: PropTypes.func.isRequired,
};