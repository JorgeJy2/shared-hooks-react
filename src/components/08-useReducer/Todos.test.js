import { shallow } from "enzyme";
import { Todos } from "./Todos";

const todos = [{
    id: 1,
    describe: 'Learn react',
    done: false
}, {
    id: 2,
    describe: 'Learn python',
    done: false
}];

describe('', ()=>{
    const wrapper = shallow(<Todos todos={todos} handlerToggle={()=>{}} handlerDelete={()=>{}} />);
    test('should snapshot', () => {
        
        expect(wrapper).toMatchSnapshot();
    });
    test('should snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test('should have two elements', () => {
        const todosHtml = wrapper.find('Todo');
        expect(todosHtml.length).toBe(todos.length);
    });

    test('should props', ()=>{
        const todoHtml = wrapper.find('Todo').at(0);
        expect(todoHtml.prop('handlerDelete')).toEqual(expect.any(Function));
        expect(todoHtml.prop('handlerToggle')).toEqual(expect.any(Function));
        expect(todoHtml.prop('index')).toBe(0);
        expect(todoHtml.prop('todo')).toEqual(todos[0]);
    })
    
});