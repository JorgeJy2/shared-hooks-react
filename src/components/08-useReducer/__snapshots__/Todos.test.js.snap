// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` should snapshot 1`] = `
<ul
  className="list-group list-group-flush"
>
  <Todo
    handlerDelete={[Function]}
    handlerToggle={[Function]}
    index={0}
    key="1"
    todo={
      Object {
        "describe": "Learn react",
        "done": false,
        "id": 1,
      }
    }
  />
  <Todo
    handlerDelete={[Function]}
    handlerToggle={[Function]}
    index={1}
    key="2"
    todo={
      Object {
        "describe": "Learn python",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;

exports[` should snapshot 2`] = `
<ul
  className="list-group list-group-flush"
>
  <Todo
    handlerDelete={[Function]}
    handlerToggle={[Function]}
    index={0}
    key="1"
    todo={
      Object {
        "describe": "Learn react",
        "done": false,
        "id": 1,
      }
    }
  />
  <Todo
    handlerDelete={[Function]}
    handlerToggle={[Function]}
    index={1}
    key="2"
    todo={
      Object {
        "describe": "Learn python",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
